text/delegatable_cdd/README.md

```diff
@@ -0,0 +1,499 @@
+level credential or even level credential depending on the issuer; an even level
+issuer always issues an odd level credential and vice versa. Each credentials in
+the credential chain is a also called a *link* and has a level which is a
+monotonically increasing number starting from 1 and has no gaps. The credential
+(or link) issued by root issuer has level 1, the subsequent link will have level
+2 and so on. During delegation (credential issuance), the delegator in addition
+to issuing a credential, sends its credential chain also to the delgatee. A
+presentation created from the delegated credential is called *attribute token*.
+
+Because verification keys of even level issuers is cryptographically different
+from verification keys of odd level issuers, verification key of even level
+issuers is called even level verification key and of odd level issuers is called
+odd level verification key. When an entity wants to be an issuer at both even
+and odd levels, it will own two keypairs, one for odd level and one for even
+level. Throughout this document and the code, the terms sigkey and verkey are
+used to refer to signing key and verification key.
+Lets look at an example. Consider an issuer chain of Alice -> Bob -> Carol ->
+Dave where Alice is the root issuer. Alice will issue a delegatable credential
+to Bob, Bob will issue a delegatable credential to Carol, Carol will issue a
+delegatable credential to Dave and Dave will create a presentation from its
+credential that will be verified by Eva. Alice is an even level issuer who has
+an even level verkey, it will issue an odd level credential with level 1 to Bob
+who has an odd level verkey and who will issue an even level credential with
+level 2 to Carol who has an even level verkey and who will issue an odd level
+credential with level 3 to Dave who has an odd level verkey. If Dave also
+interacted with an odd level issuer, he will need to have an even level verkey
+(should be with a different secret key).  
+During delegation, the delegator also signs the delegatee's verkey as one of the
+attributes of the credential. Thus the level 1 credential above (issued to Bob)
+will have Bob's odd level verkey as one of the attributes, the level 2
+credential issued to Carol will have Carol's even level verkey as one of the
+attributes and so on. When Alice delegates to Bob, she sends one credential to
+Bob, say C1, when Bob delegates to Carol, he creates a new credential C2 for
+Carol and sends C1 in addition to C2 to Carol. Similarly when Carol will issue a
+credential, she will also send C1 and C2 in addition to her issued credential.
+
+The reason for this distinction of odd and even levels is the use of *structure
+preserving cryptography*. A cryptographic scheme is called structure preserving
+if its all public inputs and outputs consist of group elements of bilinear
+groups and the functional correctness can be verified only by computing group
```

Comment:
I am quoting an external source as it is.

Reply:
If you are quoting, you should put quotes around the sentence and give a hyperlink to your source.

Reply:
I think it would be nice if you inserted a sentence somewhere near here, in which you clarify that, although the even/odd distinction is necessary to the underlying crypto, it is likely to be entirely hidden from users by intervening implementation.